### DISCUSSION TITLE
Update docs to clarify API differences from Tanstack Table

### DISCUSSION THREAD

**Discussion by githorse** [KevinVandy/material-react-table#518] [Ideas]
Just creating my first POC with MRT but already I'm a little confused by the differences between (raw) Tanstack Table and MRT. In the column def, for instance, a lot of the keys seem to be the same (`id`, `accessorKey`, `accessorFn`). Then there are some keys that clearly don't apply to (headless) Tanstack Table, like all the `mui*` options. But the relationship between the other keys is not clear -- is `Cell` identical to Tanstack's `cell`, or subtly different? (Same with `Header`/`header`, `Footer`/`footer`, etc.) 

It would definitely be helpful to me to 
1. Include a page in the docs describing the general philosophy of the relationship of MRT's APIs to Tanstack's APIs -- which kind of things are the same and which are different.
2. In each [API description](https://www.material-react-table.com/docs/api/column-options#column-options), note whether a property is the same as the one in the corresponding Tanstack API or not (or a slightly renamed version?)

By way of motivation: I'm evaluating MRT and raw Tanstack as options to replace our current datagrids. At the moment I'm working on an MRT POC and wondering, for instance, what the API of the `cell` object is. I can't find it in MRT's docs but [here's one from Tanstack](https://tanstack.com/table/v8/docs/api/core/cell) -- is it the same? I'm not sure.

**Comment by KevinVandy** [comment]
I've put some effort into color coding the links here that take you to external docs to learn more about certain props and options. Incomplete, but it's tedious work.

![image](https://github.com/KevinVandy/material-react-table/assets/28243511/9294f015-b6f5-430f-9f34-e4ba26be3c82)

If you want to writes those docs, feel free to PR, but I feel like this is stuff that can be inferred with a little research.

There are only slight api changes. Custom JSX renders are often capitalized in MRT. Lots of new props that are not in TanStack table of course.

If you really want to dig into the types, see here where they are all defined: https://github.com/KevinVandy/material-react-table/blob/main/packages/material-react-table/src/MaterialReactTable.tsx
